spec/support/helpers.rb

```diff
@@ -150,8 +152,8 @@ def gembin(cmd)
       ENV["RUBYOPT"] = old
     end
 
-    def sys_exec(cmd, expect_err = false)
-      Open3.popen3(cmd.to_s) do |stdin, stdout, stderr, wait_thr|
+    def sys_exec(cmd, expect_err = false, popen_opts = {})
+      Open3.popen3(cmd.to_s, popen_opts) do |stdin, stdout, stderr, wait_thr|
```

Comment:
you can definitely nest chdir blocks -- the issue is probably with threads or something?